src/useAsObservableSource.ts

```diff
@@ -0,0 +1,17 @@
+import { observable } from "mobx"
+import { useState } from "react"
+
+import { isPlainObject } from "./utils"
+
+export function useAsObservableSource<T>(current: T): T {
+    if (!isPlainObject(current)) {
+        throw new Error("useAsObservableSource expects an object as first argument")
+    }
+
+    const res = useState(() => observable(current, {}, { deep: false }))[0]
```

Comment:
Why not like that? I mean that `[0]` is kinda hidden there and I was confused at first. What's the benefit?
```suggestion
    const [res] = useState(() => observable(current, {}, { deep: false }))
```

Reply:
Done